### ISSUE TITLE
Strange rendering of Lua comments with "JetBrainsMono Nerd Font" when ligatures presented

### ISSUE THREAD

**Issue by daniilsvc** [neovide/neovide#1174]
**Describe the bug**
Comments in Lua files are rendered incorrectly in case of using "JetBrainsMono Nerd Font" and presence of some(!) ligatures.

**To Reproduce**

1. Create `test.lua` file with the next content:
```lua
-- I write several comment lines
-- in this file in order to demonstrate
-- effect of weird JetBrainsMono Nerd Font rendering
-- in case SOME ligaure presence.
-- Ligarute: ..
```
2. Set fonts in your config: `vim.cmd [[set guifont=JetBrainsMono\ Nerd\ Font:h12]]`
3. Run command `neovide test.lua`
4. Observe "dancing" chars:
  ![2](https://user-images.githubusercontent.com/14823534/150891805-061be9ea-cd32-48da-ab59-abff3010bf74.png)
5. Compare with file without last `..` ligature:
  ![3](https://user-images.githubusercontent.com/14823534/150893187-46d43382-3138-4a40-ad13-e1eba995fadc.png)

You can try other ligatures to reproduce these "dancing" chars. Arrors like `->` or `=>` are not producing the error. But `...`, `***`, `.=` - lead to incorrect rendering.

**Expected behavior**
No "dancing" chars. Correct rendering.

**Desktop (please complete the following information):**
 - OS: Ubuntu 20.04
 - Neovide Version 0.8.0
 - Neovim Version 0.6.1

**Please run `neovide --log` and paste the contents of the `.log` file here:**
`neovide test.lua --log` for ligature case:
[ligature.log](https://github.com/neovide/neovide/files/7929933/ligature.log)

`neovide test.lua --log` for no ligature case:
[noLigature.log](https://github.com/neovide/neovide/files/7929934/noLigature.log)

**Additional context**
I found this bug accidentally becase `..` is used in Lua for string concatenation in one of my configs files. Rendering JetBrainsMono Nerd Font in Neovide has more issues than this one. For example, if I use "dracula" colortheme (dracula/vim) with this fonts and open `pom.xml` in my project I got "dancing" chars again.
You can play around with it and find out that combination Neovide + JetBrainsMono Nerd Font + `some colorscheme` may render some elements (comments, string literals, keywords, etc.) of different file formats incorrectly.

**Comment by fredizzimo** [comment]
This is bug in the JeBrains font included in the release package of Nerdfonts, https://github.com/JetBrains/JetBrainsMono/issues/218 in combination with a lot of other height and alignment issues https://github.com/JetBrains/JetBrainsMono/releases. 

The problem is that the latest nerd fonts release is almost 2 years old, and includes version 1.0 of JetBrainsMono, which is not even available on their releases page, the first one is 1.0.1.

You can fix it by either installing the 2.2.0RC version of Nerdfonts, or build it from source.

There are probably some things that Neovide could do to render that font better, since some other editors do, but it still won't be perfect. But unless there are problems with other fonts as well, I think there are much higher priority issues to fix before that.

**Comment by daniilsvc** [comment]
Got it, thank you!
Feel free to close the issue.
